crates/net/network/src/peers.rs

```diff
@@ -807,6 +807,7 @@ impl PeersManager {
                 peer.state = PeerConnectionState::PendingOut;
                 peer.fork_id = fork_id;
                 entry.insert(peer);
+                self.connection_info.inc_pending_out();
```

Comment:
why do we need this?

Reply:
This fix is not related to this PR, but it is a bug actually.

I used `handle.connect_peer()` at first, but found that it would report an error when peer was removed. Because `self.num_pending_out` will be subtracted by 1, but it is 0 at that time, and `self.num_pending_out` is a usize type. 

For `handle.connect_peer()`, we should add 1 to `self.num_pending_out` here.

Reply:
checking, ty for the context